src/coreclr/jit/simd.h

```diff
@@ -277,6 +277,55 @@ struct simd64_t
 };
 static_assert_no_msg(sizeof(simd64_t) == 64);
 
+struct simdmask_t
+{
+    union {
+        int8_t   i8[8];
+        int16_t  i16[4];
+        int32_t  i32[2];
+        int64_t  i64[1];
+        uint8_t  u8[8];
+        uint16_t u16[4];
+        uint32_t u32[2];
+        uint64_t u64[1];
+    };
+
+    bool operator==(const simdmask_t& other) const
+    {
+        return (u64[0] == other.u64[0]);
```

Comment:
This is UB in C++ if `u64` is not the active union members, but I guess `simd64_t` has the same behavior...

Reply:
It's UB per the spec. On the other hand, it's behavior that Clang, GCC, and MSVC all use themselves and which Windows, Linux, and MacOS all have various system level defines where similar is done. So it's something that's pretty heavily relied upon for all our targets

We could always switch to defining many different structs and using std::bit_cast or reinterpet_cast, but it doesn't buy us much in practice and the current definitions roughly match the official definitions used for `__m128` and the related types used by Arm64

Reply:
I would rather be safe about stuff like this. It won't be the first time we have had bugs caused by similar cases of UB (e.g. #67669), and tracking these down is very painful. Getters/setters that use `memcpy` can be introduced to support the other typed accesses in a well-defined way.